src/Database/LSMTree/Internal/IndexCompactAcc.hs

```diff
@@ -126,26 +122,6 @@ newPinnedMVec64 lenWords = do
     setByteArray mba 0 lenWords (0 :: Word64)
     return (VUM.MV_Word64 (VPM.MVector 0 lenWords mba))
 
--- | Min\/max key-info for pages
-data Append =
-    -- | One or more keys are in this page, and their values fit within a single
-    -- page.
-    AppendSinglePage SerialisedKey SerialisedKey
-    -- | There is only one key in this page, and it's value does not fit within
-    -- a single page.
-  | AppendMultiPage SerialisedKey Word32 -- ^ Number of overflow pages
-
-instance NFData Append where
-  rnf (AppendSinglePage kmin kmax)  = rnf kmin `seq` rnf kmax
-  rnf (AppendMultiPage k nOverflow) = rnf k `seq` rnf nOverflow
-
--- | Append a new page entry to a mutable compact index.
---
--- INVARIANTS: see [construction invariants](#construction-invariants).
-append :: Append -> IndexCompactAcc s -> ST s [Chunk]
-append (AppendSinglePage kmin kmax) ica = toList <$> appendSingle (kmin, kmax) ica
-append (AppendMultiPage k n) ica        = appendMulti (k, n) ica
-
```

Comment:
Just a small suggestion for PR review ergonomics: when code is moved to a different module and subsequently changed as well, it's hard to see from the diff what the interesting code changes are. My usual approach is to move the code in one commit, and do the interesting changes in a different commit.

Right now we're only talking about a small bit of code, so it's fine and doesn't have to change in this PR, but it might be good to keep in mind for future PRs

Reply:
> My usual approach is to move the code in one commit, and do the interesting changes in a different commit.

Sounds very reasonable to me. I’ll try to remember this approach and use it in future pull requests of mine. Sadly, Git essentially doesn’t track file movement. Well, its manual page calls it right at the beginning “the stupid content tracker”. 😉

Reply:
One scenario where Git is smart enough is when a file is first moved and then its contents are changed. The diff will show that the file is renamed, instead of showing that one file was deleted and one new file was created. And the diff of the renamed file just shows the changed lines, not the unchanged lines. But that's about as smart as it gets in my experience

Reply:
Hmm, now I wonder whether this mishap might have been because of me not actually moving the file but only moving its contents. Does Git perhaps track inode numbers? 🤔

Reply:
I found the answer: Git does not track inode numbers. It detects renames based on content only. See the documentation of the `-M` option in the `git-diff(1)` man page.